src/niri.rs

```diff
@@ -332,6 +333,8 @@ pub struct Niri {
     pub hotkey_overlay: HotkeyOverlay,
     pub exit_confirm_dialog: Option<ExitConfirmDialog>,
 
+    pub pick_window: VecDeque<async_channel::Sender<Option<MappedId>>>,
```

Comment:
Since this is a modal operation, I think we should only have one in progress. Just cancel the existing one upon getting a new request.

Then also `error getting picked window info` will no longer be an error, but a regular cancellation.